lib/Doctrine/DBAL/Driver/OCI8/ExecutionMode.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Driver\OCI8;
+
+/**
+ * Encapsulates the execution mode that is shared between the connection and its statements.
+ *
+ * @internal This class is not covered by the backward compatibility promise
+ */
+final class ExecutionMode
+{
+    /** @var bool */
+    private $isAutoCommitEnabled = true;
```

Comment:
Seems to always be `true`?

Reply:
Not sure why this is introduced, if the mode is hardcoded from here on...

Reply:
> Seems to always be `true`?

What do you mean? `disableAutoCommit()` called from within `beginTransaction()` sets it to `false` unless I broke something.

Reply:
Yep, didn't read well enough, sorry! :+1: